pkg/event/event.go

```diff
@@ -259,28 +305,22 @@ connect:
 				e.UpdateClockStateMetrics(gmState, string(GM), event.IFace)
 			}
 			logOut = append(logOut, fmt.Sprintf("%s[%d]:[%s] %s T-GM-STATUS %s\n", GM, time.Now().Unix(), event.CfgName, event.IFace, gmState))
-			if lastGmState != gmState || !gmStateInitialized {
-				gmStateInitialized = true
-				glog.Infof("GM state changed from %s to %s updating clock class ", lastGmState, gmState)
-				//TODO: update clock class is going to be called for every event, this is not efficient
-				// This is going to choke the system if there are too many events
-				// since pmc call takes time, have seen 5 seconds timeout error in logs
-				err, clockClass := e.updateCLockClass(event.CfgName, gmState, event.ClockType)
-				if err == nil {
-					// in case an error do not update lastGmState so that updateCLockClass can be tried again
-					lastGmState = gmState
-					logOut = append(logOut, fmt.Sprintf("%s[%d]:[%s] CLOCK_CLASS_CHANGE %d\n", PTP4l, time.Now().Unix(), event.CfgName, clockClass))
-				} else {
-					glog.Errorf("failed to updateClockClass: %v", err)
-				}
+
+			if lastGmState != gmState && (e.clockClass == protocol.ClockClassUninitialized || e.clockClassUpdateError == nil) {
```

Comment:
should not check clockClassUpdateError here. If clockclassupdate failed before we want to try it agian.

`if lastGmState != gmState || e.clockClass == protocol.ClockClassUninitialized`

Reply:
fixed